src/contracts/GPv2Settlement.sol

```diff
@@ -137,6 +137,100 @@ contract GPv2Settlement is GPv2Signing, ReentrancyGuard, StorageAccessible {
         emit Settlement(msg.sender);
     }
 
+    /// @dev Settle a single order directly with on-chain liquidity. This
+    /// function is provided as a "fast-path" for settlements without any
+    /// coincidence of wants.
+    ///
+    /// This type of settlement always assumes the order will be filled in full,
+    /// and while it does accept partially fillable orders, they must have no
+    /// prior filled amount, and will be executed in full.
+    ///
+    /// @param order The single order to settle.
+    /// @param signingScheme The signing scheme used for the order.
+    /// @param signature The signature of the order used to recover the owner
+    /// address.
+    /// @param transfers Direct transfers of user funds to execute to interact
+    /// with on-chain liquidity.
+    /// @param interactions Smart contract interaction to perform with executed
+    /// sell amount of the order.
+    function settleOrder(
+        GPv2Order.Data memory order,
+        Scheme signingScheme,
+        bytes calldata signature,
+        GPv2AllowanceManager.DirectTransfer[] calldata transfers,
+        GPv2Interaction.Data[] calldata interactions
```

Comment:
No storageRefund?

Reply:
I figured it would accumulate and then be used as soon as there are coincidence of wants. Also, I figured gas tokens could be burned with interactions if needed.

Reply:
What's the overhead of allowing them? As far as I know we cannot yet use our own storageRefund with interactions (only gas tokens). I believe in order to account for the nonce overhead (cf. https://github.com/gnosis/gp-v2-contracts/pull/415#issuecomment-774480038) we would need it even before we have COWs.

Reply:
The overhead is probably similar to that of #428, so around 550 gas.

Reply:
Here is the result:
```
--------------+--------------
   settlement |          gas 
--------------+--------------
       w/ refunds |       161981 
      w/o refunds |       161340 
```